src/main/groovy/nextflow/processor/TaskProcessor.groovy

```diff
@@ -1793,6 +1794,19 @@ class TaskProcessor {
                     contextMap.put( param.name, val )
                     break
 
+                case FilemapInParam:
+                    def normalized = normalizeInputToFiles(val.values(),count)
+                    def mapfile = val
+                    int i = 0
+                    mapfile.keySet().each {
+                        mapfile[it] = Paths.get(normalized[i++].stageName)
+                    }
+
+                    contextMap.put( param.name,  mapfile)
+                    count += normalized.size()
+                    task.setInput(param, normalized)
+                    return
+
```

Comment:
Oh no, it is not doing it at all, forget about my part. I was wondering if then in your original code it would make sense to have some logic looking for metadata. This could be done in the FileIn case if you advise not to introduce another input/output type. And then the rest of the content of the map (meaning files) could be passed to `makeContextStage2`?